test/test_js_utils.js

```diff
@@ -131,3 +131,104 @@ describe('blocks are given IDs and can be looked up', () => {
   })
 
 })
+
+describe('blocks return proper this.columns', () => {
+
+  beforeEach(() => {
+    TidyBlocksManager.reset()
+  })
+
+  it('groupBy returns additional _group_ column', (done) => {
+    const pipeline = [
+      makeBlock( // because the pipeline has to start with a data block
+        'data_single',
+        {}),
+      makeBlock(
+        'transform_groupBy',
+        {COLUMN: 'first'})
+    ]
+    const env = evalCode(pipeline)
+    assert(env.table[0].hasOwnProperty('_group_')) // check for _group_ column
+    done()
+  })
+
+  it('ungroup removes _group_ column', (done) => {
+    const pipeline = [
+      makeBlock( // because the pipeline has to start with a data block
+        'data_single',
+        {}),
+      makeBlock(
+        'transform_groupBy', // add the _group_ column
+        {COLUMN: 'first'}),
+      makeBlock(
+        'transform_ungroup', // remove _group_ column
+        {})
+    ]
+    const env = evalCode(pipeline)
+    assert(env.table[0].hasOwnProperty('_group_') === false) // check _group_ column removed
```

Comment:
`assert(! env.table[0].hasOwnProperty('_group_'))` - remember that if you're ever doing an equality or inequality test with `true` or `false`, you can probably just use the value or its logical negation instead.

and `assert('_group_' in env.table[0])` might be more modern?